include/seastar/core/timer-set.hh

```diff
@@ -151,15 +160,28 @@ public:
      * Postconditions:
      *  - timer is no longer in the active set.
      *  - this object will no longer hold any references to this timer.
+     *
+     * get_next_timeout() will return correct next time point after removing
+     * timer with earliest timeout.
      */
     void remove(Timer& timer) noexcept
     {
+        // The code below very likely to have a bug in detemining correct bucket
+        // actual bucket shall be in range index - bucket_n
+        // timer might be not found in real storing bucket, _non_empty_buckets
+        // might have invalid state
```

Comment:
I don't understand this comment. If it has a bug, can you fix it? If it doesn't, we shouldn't say it is "very likely to have a bug"...

Reply:
@nyh I didn't wanted to merge this as it is. As proper and effective fix seems not to be really straight forward. At first I wanted some comment from the support side.

Reply:
Comment has been removed. Will add separated branch with asserts to highlight issue if exists.